docs/src/examples.rst

```diff
@@ -14,205 +14,216 @@
 
-   code-block:: js
+  In addition, the |cdk| will prevent instances from being deleted when
+  autoscaling tries to kill an instance,
+  but either a task is running or is scheduled on that instance.
 
-   import { App } from '@aws-cdk/cdk'
-   import { MyStack } from './my-stack'
-   import { DeploymentPipeline } from './my-deployment'
+  Previously, you had to create a Lambda function to have this functionality.
+  
+* Asset support, so that you can deploy source from your machine to |ECS| in one step
+  Previously, to use application source you had to perform a number of manual steps
+  (upload to |ECR|, create Docker image, etc.).
+ 
+.. _creating_ecs_l2_example_1:
 
-   const app = new App();
+Step 1: Create the Directory and Initialize the |cdk|
+-----------------------------------------------------
 
-   // Use the default environment
-   new MyStack(app, { name: 'Dev' });
+Let's start with creating a new directory to hold our |cdk| code
+and create a new app in that directory.
 
-   // Pre-production stack
-   const preProd = new MyStack(app, {
-     name: 'PreProd',
-     env: { region: 'us-west-2' },
-     preProd: true
-   });
+.. code-block:: sh
 
-   // Production stacks
-   const prod = [
-     new MyStack(app, {
-       name: 'NAEast',
-       env: { region: 'us-east-1' }
-   }),
+    mkdir MyEcsConstruct
+    cd MyEcsConstruct
 
-   new MyStack(app, {
-     name: 'NAWest',
-     env: { region: 'us-west-2' }
-   }),
+.. tabs::
 
-   new MyStack(app, {
-     name: 'EU',
-     env: { region: 'eu-west-1' },
-       encryptedStorage: true
-     })
-   ]
+    .. group-tab:: TypeScript
 
-   // CI/CD pipeline stack
-   new DeploymentPipeline(app, {
-     env: { region: 'us-east-1' },
-      strategy: DeploymentStrategy.Waved,
-      preProdStages: [ preProd ],
-      prodStages: prod
-   });
+        .. code-block:: sh
 
-   app.run();
+            cdk init --language typescript
 
-.. _dynamodb_example:
+        Update *my_ecs_construct.ts* in the *bin* directory to only contain the following code:
 
-Creating a |DDB| Table
-======================
+        .. code-block:: ts
 
-The following example creates a
-|DDB| table with the partition key **Alias**
-and sort key **Timestamp**.
+            import cdk = require('@aws-cdk/cdk');
 
-.. code-block:: js
+            class MyEcsConstructStack extends cdk.Stack {
+              constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
+                super(parent, name, props);
 
-   import dynamodb = require('@aws-cdk/aws-dynamodb');
-   import cdk = require('@aws-cdk/cdk');
 
-   class MyStack extends cdk.Stack {
-     constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
-       super(parent, name, props);
+              }
+            }
 
-       const table = new dynamodb.Table(this, 'Table', {
-         tableName: 'MyAppTable',
-         readCapacity: 5,
-         writeCapacity: 5
-       });
+            const app = new cdk.App();
```

Comment:
In fact these lines will also be gone.

These lines will live in a file in the `bin` directory, the other lines will live in a file in the `lib` directory.

Here's the new version of the *app* template: https://github.com/awslabs/aws-cdk/tree/master/packages/aws-cdk/lib/init-templates/app/typescript